drivers/SmartThings/bose/src/init.lua

```diff
@@ -300,6 +300,28 @@ local bose = Driver("bose", {
   },
 })
 
+local function ip_change_check()
+  local id_to_device = {}
+  local device_list = bose:get_devices()
+  for _, device in ipairs(device_list) do
+    local id = bose_utils.get_serial_number(device)
+    id_to_device[id] = device
+  end
+  discovery.find(nil, function(found)
+    local known = id_to_device[found.id]
+    if known ~= nil then
+      local known_ip = known:get_field("ip")
+      if known_ip == nil or known_ip ~= found.ip then
+        log.trace("Updating device ip:", found.id, found.ip)
+        known:set_field("ip", found.ip, { persist = true })
+      end
+    end
+  end)
+end
+
+local IP_CHANGE_CHECK_INTERVAL_S = 60
```

Comment:
I think for Hue and Sonos we defaulted to 10m instead of 1m, I'm not sure which makes more sense here tbh though. I feel like there's value to being very responsive but there's also value in reducing LAN traffic and hub load. Not a nit or a blocker from my perspective, but wanted to make a comment to have a discussion; no pressure to actually incorporate this change.

```suggestion
local IP_CHANGE_CHECK_INTERVAL_S = 600
```

Reply:
I think hue is also at a minute right now https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/philips-hue/src/init.lua#L1448 

I wasn't 100% on this being a minute so I have no problem changing this to be 10 minutes.

Reply:
Womp, I should fix that. I thought we did that already.